Solutions/Digital Shadows/Data Connectors/Digital Shadows/DigitalShadowsConnectorAzureFunction/AS_api.py

```diff
@@ -0,0 +1,68 @@
+""" handles all Azure sentinel apis related functions here """
+import base64
+import datetime
+import requests
+import hmac
+import hashlib
+import logging
+
+logger = logging.getLogger("AS_api")
+
+class logs_api:
+    """ 
+    class for log analytics api 
+    """
+    
+    def __init__(self, id, key):                            
+        """ 
+            constructor initializing azure creds.
+            id is workspace id and key is primary key
+        """
+        
+        self.customer_id = id
+        self.shared_key = key
+
+    
+    def build_signature(customer_id, shared_key, date, content_length, method, content_type, resource):
```

Comment:
@shikhin-metron - this method should either have the `@staticmethod` decorator (since you are accessing it as `logs_api.build_signature`)

or you add `self` as the first parameter and change how you are calling it to be `self.build_signature`